### ISSUE TITLE
Run ESLint and Typecheck on CI and use pnpm

### ISSUE THREAD

**Issue by andresgutgon** [IdoPesok/zsa#137]
# What?
Not really an issue. But a piece of feedback 😃 

## CI lint & Typecheck
While working in this PR https://github.com/IdoPesok/zsa/pull/124 I noticed CI is not running Linter or Typescript checks. 

Is there any reason for that?

Also notice ESLint config is a big convoluted with double `tsconfig.json` and `tsconfig.lint.json` in all the packages. I think this could be simplified. 

## Use PNPM instead of NPM
I admit this one is not critical 😂 . But I've working in all my monorepos with pnpm is so much better experience. Faster builds and better module resolution.

**Comment by IdoPesok** [comment]
Good catch on the checks, just added `turbo typecheck` to the CI. Will work on running the linter.

With regards to the ESLint config, this is what turborepo recommends in their [examples](https://github.com/vercel/turbo/tree/main/examples/basic/packages/ui)

Going to move this issue to feedback category in the new discussions page.
